src/libAtomVM/posix_nifs.c

```diff
@@ -362,14 +362,17 @@ static term nif_atomvm_posix_read(Context *ctx, int argc, term argv[])
     VALIDATE_VALUE(count_term, term_is_integer);
     int count = term_to_int(count_term);
 
+    size_t size = term_binary_data_size_in_terms(count) + BINARY_HEADER_SIZE + TERM_BOXED_SUB_BINARY_SIZE + TUPLE_SIZE(2);
+    if (UNLIKELY(memory_ensure_free_with_roots(ctx, size, 2, argv, MEMORY_CAN_SHRINK) != MEMORY_GC_OK)) {
```

Comment:
We do not need to keep argv[1] in roots AFAIK

Reply:
Yup, but I thought it would be more future-proof to put all the args in the roots, and it shouldn't have noticeable performance overhead